Refactor icons-svg@4.x with gulp

HeskeyBaozi opened the pull request:
此 pr 包含以下可能具有破坏性的改动：

## 包相关变动

`packages/icons` 中的包名由 `@ant-design/icons-svg` 修改为 `@ant-design/icons-svg-legacy`。包名 `@ant-design/icons-svg` 由项目下的目录重写的 `packages/icons-svg` 继承。

新的 `@ant-design/icons-svg` 包和原来的版本 ( `4.0.0-alpha.0` 和 `@ant-design/icons <= 2.1.1` ) 相比，有如下改动：

- 不再提供用于存放未处理的图标的 `svg` 目录
- 不再提供用于全量引入的 `dist.js` 文件和压缩过的 `umd.js` 文件，全量引入直接用 `import * as allIcons from '@ant-design/icons-svg'`， 走 `index.js` 即可
- 不再提供用于 `tree-shaking` 的 `lib/index.es.js` 文件
- 不再提供用于字符串字面量引入用的短横线图标名称的清单 `manifest.js` 文件
- 提供 `es` 目录用于 `tree-shaking`

最终用户得到的目录如下：

```yml
es
inline-svg # 此为处理后存放的svg图标
lib
package.json
ReadMe.md
```

这些改动的原则都是倾向于按需引入，保证主入口文件只有图标的导出，使得 `import * as allIcons from '@ant-design/icons-svg'` 只会导出图标信息。

### 此PR来源于
https://github.com/ant-design/ant-design-icons/pull/90
`@ant-design/icon-kit` 这个脚手架已自我废弃

### 目前图标

[查看当前全部图标](https://github.com/HeskeyBaozi/ant-design-icons/blob/refactor/icons-svg-4.x-glup/packages/icons-svg/docs/list.md)

### 为什么

- 不使用 `rxjs`
响应式编程这类范式强大之处在于对可观察数据的时序性控制和数学上持续性赋值`:=`不同流的可组合性。图标生成就是文件系统IO和数据结构转换。显然`gulp`及其生态能把这个任务做得更好，`rxjs`也可以做但显然其长处并没有发挥出来，且编写无确切规范导致维护性差。

- 速度与维护选择

| Before | icon-kit (rxjs) | gulp |
|:-----:|:----:|:---:|
![old-generate-use-15s](https://user-images.githubusercontent.com/15819224/62518030-298aae80-b85b-11e9-9195-e34ee8f3173b.png)|![new](https://user-images.githubusercontent.com/15819224/62518041-30192600-b85b-11e9-8ff8-290e5207eddd.png)|![after](https://user-images.githubusercontent.com/15819224/62720794-5f878880-ba3d-11e9-80dc-062ead439193.jpg)
| 15.42s | 6.88s | 9.55s |


### WIP

- [x] npm prepublish

HeskeyBaozi committed:
use gulp

HeskeyBaozi committed:
add svg

HeskeyBaozi committed:
init gulp file

HeskeyBaozi committed:
clean task

HeskeyBaozi committed:
fix focusable & viewBox

HeskeyBaozi committed:
rename

HeskeyBaozi committed:
stringify

HeskeyBaozi committed:
use constant

HeskeyBaozi committed:
inline SVG

HeskeyBaozi committed:
use concat

HeskeyBaozi committed:
generate index.ts

HeskeyBaozi committed:
use tasks

HeskeyBaozi committed:
generate list

HeskeyBaozi committed:
use father

HeskeyBaozi committed:
add docs list

HeskeyBaozi committed:
add test

HeskeyBaozi committed:
update readme

HeskeyBaozi committed:
add prepublish

HeskeyBaozi committed:
add -legacy suffix

HeskeyBaozi committed:
twotone -> twoTone

HeskeyBaozi committed:
icons list added to readme.md

HeskeyBaozi committed:
twotone -> twoTone in index.js

vagusX committed:
refactor/warning in icons (#92)

* chore: cleanup

* feat: update scripts for publish usage

* feat: generate icons

* feat: use rc-util/lib/warning

* scripts: add np for release usage

vagusX committed:
fix: types for auto import (#93)

vagusX committed:
4.0.0-alpha.4

vagusX committed:
fix: insert css only in browser (#94)

vagusX committed:
4.0.0-alpha.5

vagusX committed:
fix: clean up d.ts (#95)

vagusX committed:
4.0.0-alpha.6

vagusX committed:
update build scripts to support real tree shaking (#96)

* fix: add .storybook folder to npm ignore

* fix: warning msg in icons

* build: update scripts

* feat: rebuild icons

vagusX committed:
4.0.0-alpha.7

vagusX commented:
@HeskeyBaozi 这个 mr 合并了，icon-vue/ng 也要跟进下吧？老的 @ant-design/icons-svg-legacy folder 后续考虑删掉

dependabot-preview[bot] committed:
build(deps-dev): update @types/jest requirement from ^23.3.1 to ^24.0.17

Updates the requirements on [@types/jest](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/jest) to permit the latest version.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/jest)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

HeskeyBaozi commented:
https://github.com/ant-design/ant-design-icons/pull/100
